### PR TITLE
fix: [CHAOS-1999]: Fixing Service Discovery Table Data and missing API calls

### PR DESCRIPTION
**Pull request by S-ayanide**
### Summary

Adds the following changes
* Add blacklisted namespace to create dAgent API from FE
* Discovery agent delete api not integrated
* Show audit info(last updated by and time , created by as per design) for discovery agent
* Show audit info for network map (last updated by and time , created by as per design)
* Network map delete api not integrated

#### Screenshots

Screenshot 2023-07-06 at 3 47 38 PM
Screenshot 2023-07-05 at 4 42 18 PM
Screenshot 2023-07-05 at 4 38 06 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by SahilKr24 — APPROVED** [review]

**Review by hrishavjha — DISMISSED** [review]
LGTM

**Review by arkajyotiMukherjee — DISMISSED** [review]

File: src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx [hunk 1]
```diff
@@ -34,12 +36,17 @@ import { useStrings } from 'framework/strings'
 import EmptyStateNetworkMap from './EmptyStateNetworkMap'
 import css from './NetworkMapTable.module.scss'
 
-const NetworkMapTable: React.FC = /* istanbul ignore next */ () => {
+interface NetworkMapTableProps {
+  agentName: string | undefined
+  connectorID: string | undefined
+}
+
+const NetworkMapTable: React.FC<NetworkMapTableProps> = /* istanbul ignore next */ ({ agentName, connectorID }) => {
```
Comment (abhinavrastogi-harness) [inline src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx:44]: we should not ignore entire components. why do we need the ignore here?

File: src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx [hunk 2]
```diff
@@ -103,32 +106,78 @@ const NetworkMapTable: React.FC = /* istanbul ignore next */ () => {
...
           <Menu data-testid="options" style={{ backgroundColor: 'unset' }}>
-            <MenuItem icon="edit" text={'Edit'} onClick={() => void 0} className={css.menuItem} />
-            <MenuItem
-              icon="delete"
-              text={'Delete'}
-              className={css.menuItem}
-              onClick={() => handleDelete(row.original.id ?? '')}
-            />
+            <MenuItem icon="edit" text={'Edit'} disabled onClick={() => void 0} className={css.menuItem} />
+            <MenuItem icon="trash" text={getString('delete')} className={css.menuItem} onClick={handleDelete} />
```
Comment (abhinavrastogi-harness) [inline src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx:180]: edit and delete should be rbac-controlled

File: src/modules/73-discovery/pages/home/views/create-discovery-agent/CreateDAgent.tsx [hunk 3]
```diff
@@ -66,18 +68,32 @@ const CreateDAgent: React.FC<DrawerProps> = /* istanbul ignore next */ ({ setDra
...
+    minutes: 15,
+    blacklistedNamespaces: ['kube-node-lease', 'kube-public', 'kube-system'],
     detectNetworkTrace: isNetworkTraceDetected
   }
 
   const handleSubmit = (): void => {
     if (dAgentFormRef.current) {
       dAgentFormRef.current.validateForm().then(errors => {
-        if (Object.keys(errors).length === 0) {
+        if (dAgentFormRef.current?.values.minutes && dAgentFormRef.current.values.minutes < 15) {
```
Comment (abhinavrastogi-harness) [inline src/modules/73-discovery/pages/home/views/create-discovery-agent/CreateDAgent.tsx:81]: why aren't we using `yup` validations for this?

File: src/modules/73-discovery/pages/home/views/create-discovery-agent/CreateDAgent.tsx [hunk 4]
```diff
@@ -136,7 +152,8 @@ const CreateDAgent: React.FC<DrawerProps> = /* istanbul ignore next */ ({ setDra
                 .max(50, getString('discovery.dAgentValidation.nameStart'))
                 .required(getString('discovery.dAgentValidation.nameRequired')),
               connectorRef: Yup.string().trim().required(getString('discovery.dAgentValidation.connectConnector')),
-              discoveryNamespace: Yup.string().trim().required(getString('discovery.dAgentValidation.selectNamespace'))
+              discoveryNamespace: Yup.string().trim().required(getString('discovery.dAgentValidation.selectNamespace')),
+              duration: Yup.number().required(getString('discovery.dAgentValidation.durationRequired'))
```
Comment (abhinavrastogi-harness) [inline src/modules/73-discovery/pages/home/views/create-discovery-agent/CreateDAgent.tsx:156]: you can set min and max also here for numbers

File: src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx (hunk 1 above)
Comment (S-ayanide) [reply src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx:44]: We are unable to get the branch coverage for this component above 55% so as a last resort we have ignored it, @arkajyotiMukherjee is parallely working on resolving this

File: src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx (hunk 2 above)
Comment (S-ayanide) [reply src/modules/73-discovery/pages/discovery-details/views/network-map/NetworkMapTable.tsx:180]: Yes, RBAC fixes are being raised here (https://github.com/harness/harness-core-ui/pull/16773)

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by SahilKr24 — APPROVED** [review]

**Review by hrishavjha — APPROVED** [review]

**Pull request merged**